src/genegraph/migration.clj

```diff
@@ -163,3 +169,28 @@
       (retrieve-migration env/genegraph-bucket archive-file env/data-vol)
       (decompress-database env/data-vol archive-path))))
 
+(defn load-stream-data
+  "Loads stream data from scratch into an existing database"
+  ([dest-path] (load-stream-data dest-path {}))
+  ([dest-path {from-scratch :from-scratch}]
+   (log/info :fn :load-stream-data :msg (str "Loading stream data into database at " dest-path))
+   (with-redefs [env/data-vol dest-path]
+     (stop #'event/stream-processing)
+     (populate-data-vol-if-needed)
+     ;(fs/mkdirs env/data-vol)
+     (start #'db/db)
+     (start #'property-store/property-store)
+
+     (log/info :fn :load-stream-data :msg "Resetting topic offsets...")
+     (if from-scratch
+       (do (base/initialize-db!)
+           (batch/process-batched-events!)
+           (fs/delete (stream/offset-file))))
+     (stream/initialize-current-offsets!)
+     (start #'event/stream-processing)
+     (log/info :fn :load-stream-data :msg "Processing streams...")
+     (stream/wait-for-topics-up-to-date)
+     (log/info :fn :load-stream-data :msg "Stopping streams...")
+     (stop #'event/stream-processing)
+     (log/info :fn :load-stream-data :msg "Waiting for streams to close...")
+     (stream/wait-for-topics-closed))))
```

Comment:
This code repeats existing code, creating some messiness it would be good to avoid. It seems like this is a bit of a patch that lets you play data on top of an existing migration; maybe we could have a discussion about what we want out of this and the best approach to follow.